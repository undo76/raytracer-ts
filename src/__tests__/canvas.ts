import {
  color,
} from '../tuple';

import Canvas from '../canvas';

describe('Canvas', () => {
  const myCanvas = new Canvas(10, 20);
  const black = color(0, 0, 0);
  const blue = color(0, 0, 1);

  test('Dimensions are set', () => {
    expect(myCanvas.width).toEqual(10);
    expect(myCanvas.height).toEqual(20);
  });

  test('Every pixel is black by default', () => {
    expect(myCanvas.pixelAt(0, 0)).toEqual(black);
    expect(myCanvas.pixelAt(5, 5)).toEqual(black);
  });

  test('Setting a color', () => {    
    myCanvas.setPixel(3, 5, blue);
    expect(myCanvas.pixelAt(0, 0)).toEqual(black);
    expect(myCanvas.pixelAt(3, 5)).toEqual(blue);
  });
});
