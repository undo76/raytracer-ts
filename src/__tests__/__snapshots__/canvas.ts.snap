// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas PPM conversion 1`] = `
"P3
4 3
255
255 25 0 255 25 0 255 25 0 255 25 0 255 25 0 255 25 0 255 25 0 255 25 0 255 25 0 255 25 0
255 25 0 255 25 0
"
`;
